src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -331,6 +345,20 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun pluralise(count: Int, singular: String, plural: String): String =
         if (count == 1) singular else plural
 
+    private fun canElideReturnLifetime(fn: PsiElement): Boolean {
```

Comment:
And [here's the rustc code](https://github.com/rust-lang/rust/blob/0f49616a53465d3ee7a7069cd13c7f86a483b655/src/librustc/middle/resolve_lifetime.rs#L1275).